R/Seurat_III.R

```diff
@@ -194,126 +194,260 @@ MarkStepRun <- function(seuratObj, name, saveFile = NULL) {
     seuratObj <- MarkStepRun(seuratObj, 'NormalizeData')
-  }
-  else {
-    for (exptNum in nameList) {
-      if (is.null(seuratObj)) {
-        seuratObj <- seuratObjs[[exptNum]]
-      } else {
-        seuratObj <- merge(x = seuratObj,
-        y = seuratObjs[[exptNum]],
-        project = projectName,
-        do.normalize = F)
-      }
+    
+    return(seuratObj)
+    
+  } else {
+      warning("only one file in list. no need to merge. returning the single object")
+      return(seuratObjs[[1]])
     }
+}
 
-    print('after merge')
-    print(seuratObj)
+
+#' @title doMergeSimple
+#' @description An internal method to do simple merging of seurat objects from a list of them.
+#' @param seuratObjs A list of seurat objects, optionally named (in which case these will be used as dataset names). Also can use SplitObject(, split.by =)
+#' @param nameList A list of names from MergeSeuratObjs()
+doMergeSimple <- function(seuratObjs, nameList, projectName){
+  
+  seuratObj <- NULL
+  
+  for (exptNum in nameList) {
+    print(exptNum)
+    if (is.null(seuratObj)) {
+      seuratObj <- seuratObjs[[exptNum]]
+    } else {
+      seuratObj <- merge(x = seuratObj,
+                         y = seuratObjs[[exptNum]],
+                         project = projectName)
+    }
+    
   }
+  return(seuratObj)
+}
+
+
+
+#' @title MergeSeuratObjs
+#' @description Merges a list of Seurat objects, using Seurat::IntegrateData()
+#' @param seuratObjs A list of seurat objects, optionally named (in which case these will be used as dataset names). Also can use SplitObject(, split.by =)
+#' @param metadata A list of metadata.  If provided, the names of this list will be used as dataset names
+#' @param alignData If true, data will be aligned using Seurat::IntegrateData()
+#' @param maxCCAspaceDim The number of dims to use with FindIntegrationAnchors()
+#' @param maxPCs2Weight The number of dims to use with IntegrateData()
+#' @param projectName The project name when creating the final seuratObj
+#' @param useAllFeatures If true, the resulting object will contain all features, as opposed to just VariableGenes (not recommended)
+#' @param nVariableFeatures The number of VariableFeatures to identify
+#' @param includeCellCycleGenes If true, the cell cycles genes will always be included with IntegrateData(), as opposed to just VariableGenes
+#' @return A modified Seurat object.
+#' @export
+#' @importFrom methods slot
+MergeSeuratObjs <- function(seuratObjs, metadata=NULL, 
+                            alignData = T, 
+                            method = "simple", #"cca" or etc
+                            maxCCAspaceDim = 20, maxPCs2Weight = 20, 
+                            useAllFeatures = F, nVariableFeatures = 2000,
+                            includeCellCycleGenes = T, assay = NULL,
+                            normalization.method = "LogNormalize", 
+                            spike.genes = NULL){
+  
+  # #Testing with panc8 data
```

Comment:
This is useful, but I recommend moving this into an actual test.  you can add a new one in test-mergeseurat.R